lib/perl/Genome/Model/Build/SingleSampleGenotype.pm

```diff
@@ -71,4 +73,48 @@ sub reference_being_replaced_for_input {
     return;
 }
 
+sub symlink_results {
+    my $self = shift;
+    my $destination = shift;
+
+    my $top_directory = File::Spec->join($destination, $self->id);
+    Genome::Sys->create_directory($top_directory);
+
+    my $alignment = $self->merged_alignment_result;
+    my $alignment_dir = File::Spec->join($top_directory, 'alignments');
+    Genome::Sys->create_symlink($alignment->output_dir, $alignment_dir);
+
+    my $qc = $self->qc_result;
+    my $qc_dir = File::Spec->join($top_directory, 'qc');
+    Genome::Sys->create_symlink($qc->output_dir, $qc_dir);
+
+    my $variant_dir = File::Spec->join($top_directory, 'variants');
+    Genome::Sys->create_directory($variant_dir);
+
+    for my $hc_result ($self->haplotype_caller_result) {
+        my $hc_basename = Genome::Utility::Text::sanitize_string_for_filesystem(
+            join('-', $hc_result->intervals, $hc_result->id)
+        );
+        my $hc_dir = File::Spec->join($variant_dir, $hc_basename);
+        Genome::Sys->create_symlink($hc_result->output_dir, $hc_dir);
+    }
+
+    return $top_directory;
+}
+
```

Comment:
I didn't do exactly what was suggested here, but I do think breaking this down a little made the code easier to follow.  Thanks for the discussion!

Some history (some of which is also recording yesterday's discussion): The name `create_symlinks` was taken for parallelism with the already-named feature in `Genome::Process` (which also used the duck-typing approach that this PR has now changed away from).  But it really creates a simulated build directory--the names of the paths and the contents in the directory are arbitrary at the discretion of a particular build type.  Most build types do delegate in order to fill in the contents of particular directories (notably to `DetectVariants2` for the `variants` directory), but SSG is the only thing using `HaplotypeCaller` results for now.